uaclient/entitlements/base.py

```diff
@@ -60,6 +62,12 @@ class UAEntitlement(metaclass=abc.ABCMeta):
     # List of services that are incompatible with this service
     _incompatible_services = []  # type: "List[str]"
 
+    # List of services that must be active before enabling this service
+    _required_services = []  # type: "List[str]"
```

Comment:
It's probably safer for us to use immutables like a tuple for these class attributes because if we grow methods that manipulate these class attrs in this future in subclasses, isn't it possible that the mutable list would be overridden for all subclasses?

Reply:
Since we are actually using a property to read those the values from these variables, I think this problem can still happen. However, I agree we should change them to a tuple regardless